[Conclusions] the human factor

enricoferrero opened the pull request:
As discussed in #375, I added a paragraph to the conclusions to emphasise the importance of the human factor. Feedback on content, style and grammar very welcome.

I also added a short explanation of why I (we?) believe unsupervised methods might have most potential - please let me know if you agree/disagree with the changes.

Finally, I added a 'how' in the last sentence because I thought it was missing but I'm not 100% sure I understood what @cgreene was trying to say - please have a look.

cgreene commented:
This looks good to me. 👍 I think the unsupervised bit is saying: `here's a way we'll know there's been a transformation` and not `here's what I expect to see tomorrow`, so that is consistent with @akundaje's feedback too.

agitter commented on sections/07_conclusions.md:
```diff
@@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
+a number of biomedical domains, and yet we do not believe that biologists and
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the
```
How about this rephrasing:
> better than scientists at specific tasks, in many cases they would still be unable to fully grasp the bigger picture

to

> better than scientists at some specific tasks, they may not fully grasp the bigger picture.

agitter commented on sections/07_conclusions.md:
```diff
@@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
+[@arxiv:1703.06857], an important reminder that these algorithms do not really
+understand or "recognise the semantic of the object". Until true and reliable
+artificial intelligence becomes standard in the laboratory and in the clinic,
+the human factor still has a critical role to play. In some cases, cooperation
```
I like raising this human factor in conclusion.  I tried to present this idea in the revised intro as well.

agitter commented on sections/07_conclusions.md:
```diff
@@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
+bigger picture. Deep neural networks are also prone to mistakes that humans are
+much more unlikely to make, such as incorrectly classifying negative images
+[@arxiv:1703.06857], an important reminder that these algorithms do not really
+understand or "recognise the semantic of the object". Until true and reliable
```
"semantics", and we could drop the quotes around this phrase.

agitter commented on sections/07_conclusions.md:
```diff
@@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the
+bigger picture. Deep neural networks are also prone to mistakes that humans are
```
You've been following the discussion in #384, and I agree that this is a more solid way to make this point.  We can replace "such as incorrectly classifying negative images [@arxiv:1703.06857]" with something like "such as susceptibility to adversarial examples [a couple references from #384]".

In the next sentence, you can acknowledge defensive techniques and then continue with the point you want to make: "Despite progress in addressing specific limitations such as adversarial inputs [a couple references from #384 or Section 2.3 of the paper I linked], until true and reliable..."

agitter commented on sections/07_conclusions.md:
```diff
@@ -64,11 +81,13 @@ transcription factor binding prediction and molecular graphs instead of
-transformed biology and medicine.
+currently less developed than their supervised counterparts, but they may have
+the most potential because of how expensive and time-consuming it is to label
+the large amounts of unlabelled data existing in the field of biomedicine. When
```
Could shorted to "label the large amounts of biomedical data".

akundaje commented:
Btw, I think it's important to mention that adversarial examples are a
problem for all ML methods. It's not specific to deep learning.

http://www.kdnuggets.com/2015/01/deep-learning-flaws-universal-machine-learning.html

On May 7, 2017 5:38 AM, "Anthony Gitter" <notifications@github.com> wrote:

*@agitter* requested changes on this pull request.

Looks good to me. My main comment pertains to our discussion in #384
<https://github.com/greenelab/deep-review/issues/384>.
------------------------------

In sections/07_conclusions.md
<https://github.com/greenelab/deep-review/pull/385#discussion_r115141490>:

> @@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
 for some problems, for instance, predicting gene regulation in disease, perfect
 accuracy will be unattainable.

+We are witnessing deep learning models achieving human-level performance across
+a number of biomedical domains, and yet we do not believe that biologists and
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the

How about this rephrasing:

better than scientists at specific tasks, in many cases they would still be
unable to fully grasp the bigger picture

to

better than scientists at some specific tasks, they may not fully grasp the
bigger picture.

------------------------------

In sections/07_conclusions.md
<https://github.com/greenelab/deep-review/pull/385#discussion_r115141561>:

> @@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
 for some problems, for instance, predicting gene regulation in disease, perfect
 accuracy will be unattainable.

+We are witnessing deep learning models achieving human-level performance across
+a number of biomedical domains, and yet we do not believe that biologists and
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the
+bigger picture. Deep neural networks are also prone to mistakes that humans are
+much more unlikely to make, such as incorrectly classifying negative images
+[@arxiv:1703.06857], an important reminder that these algorithms do not really
+understand or "recognise the semantic of the object". Until true and reliable
+artificial intelligence becomes standard in the laboratory and in the clinic,
+the human factor still has a critical role to play. In some cases, cooperation

I like raising this human factor in conclusion. I tried to present this
idea in the revised intro as well.
------------------------------

In sections/07_conclusions.md
<https://github.com/greenelab/deep-review/pull/385#discussion_r115141598>:

> @@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
 for some problems, for instance, predicting gene regulation in disease, perfect
 accuracy will be unattainable.

+We are witnessing deep learning models achieving human-level performance across
+a number of biomedical domains, and yet we do not believe that biologists and
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the
+bigger picture. Deep neural networks are also prone to mistakes that humans are
+much more unlikely to make, such as incorrectly classifying negative images
+[@arxiv:1703.06857], an important reminder that these algorithms do not really
+understand or "recognise the semantic of the object". Until true and reliable

"semantics", and we could drop the quotes around this phrase.
------------------------------

In sections/07_conclusions.md
<https://github.com/greenelab/deep-review/pull/385#discussion_r115141741>:

> @@ -52,6 +52,23 @@ addition, the stochasticity and complexity of biological systems implies that
 for some problems, for instance, predicting gene regulation in disease, perfect
 accuracy will be unattainable.

+We are witnessing deep learning models achieving human-level performance across
+a number of biomedical domains, and yet we do not believe that biologists and
+clinicians will be out of a job anytime soon. While deep learning and other
+advanced algorithms will soon be (or already are) better than scientists at
+specific tasks, in many cases they would still be unable to fully grasp the
+bigger picture. Deep neural networks are also prone to mistakes that humans are

You've been following the discussion in #384
<https://github.com/greenelab/deep-review/issues/384>, and I agree that
this is a more solid way to make this point. We can replace "such as
incorrectly classifying negative images [@arxiv
<https://github.com/arxiv>:1703.06857]"
with something like "such as susceptibility to adversarial examples [a
couple references from #384
<https://github.com/greenelab/deep-review/issues/384>]".

In the next sentence, you can acknowledge defensive techniques and then
continue with the point you want to make: "Despite progress in addressing
specific limitations such as adversarial inputs [a couple references from
#384 <https://github.com/greenelab/deep-review/issues/384> or Section 2.3
of the paper I linked], until true and reliable..."
------------------------------

In sections/07_conclusions.md
<https://github.com/greenelab/deep-review/pull/385#discussion_r115141757>:

> @@ -64,11 +81,13 @@ transcription factor binding prediction and molecular graphs instead of
 pre-computed bit vectors for drug discovery. These flexible input feature
 representations have spurred creative modeling approaches that would be
 infeasible with other machine learning techniques. Unsupervised methods are
-currently less-developed than their supervised counterparts, but they may have
-the most potential. When deep learning algorithms can summarize very large
-collections of input data into interpretable models that spur scientists to ask
-questions that they didn't know to ask, it will be clear that deep learning has
-transformed biology and medicine.
+currently less developed than their supervised counterparts, but they may have
+the most potential because of how expensive and time-consuming it is to label
+the large amounts of unlabelled data existing in the field of biomedicine. When

Could shorted to "label the large amounts of biomedical data".

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/greenelab/deep-review/pull/385#pullrequestreview-36663347>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAI7EUXKpya0g8TCx9D0B9rDNiwUW5T-ks5r3btDgaJpZM4NS38h>
.

enricoferrero commented:
Thanks @akundaje @agitter @cgreene for the great feedback here and in #384.
I *think* I have now addressed all the points you made - please have a look and let me know.

agapow commented on sections/07_conclusions.md:
```diff
@@ -52,8 +52,26 @@ addition, the stochasticity and complexity of biological systems implies that
 for some problems, for instance, predicting gene regulation in disease, perfect
 accuracy will be unattainable.
 
+We are witnessing deep learning models achieving human-level performance across
```
Just to get rid of duplicate "we": maybe "a number of biomedical domains, yet this does not mean biologists or clinicians will be out of a job anytime soon."

enricoferrero replied:
Done

agapow commented on sections/07_conclusions.md:
```diff
@@ -52,8 +52,26 @@ addition, the stochasticity and complexity of biological systems implies that
+fully grasp the bigger picture. Machine learning algorithms, including deep
+neural networks, are also prone to mistakes that humans are much less likely to
+make, such as misclassification of adversarial examples [@arxiv:1312.6199
+@arxiv:1412.6572], a reminder that these algorithms do not necessarily
```
"do not necessarily understand" to "do not understand"?

agapow commented on sections/07_conclusions.md:
```diff
@@ -52,8 +52,26 @@ addition, the stochasticity and complexity of biological systems implies that
+neural networks, are also prone to mistakes that humans are much less likely to
+make, such as misclassification of adversarial examples [@arxiv:1312.6199
+@arxiv:1412.6572], a reminder that these algorithms do not necessarily
+understand the semantics of the objects presented. Despite progress in
```
Maybe, because there's arguably only one limitation described: "Despite some progress in
addressing this limitation"

agapow commented on sections/07_conclusions.md:
```diff
@@ -52,8 +52,26 @@ addition, the stochasticity and complexity of biological systems implies that
+true and reliable artificial intelligence becomes standard in the laboratory and
+in the clinic, the human factor still has a critical role to play. In some
+cases, cooperation between human experts and deep learning algorithms can
+achieve better performance than either of them individually [@arxiv:1606.05718].
```
Delete "of them"

agapow commented on sections/07_conclusions.md:
```diff
@@ -64,11 +82,12 @@ transcription factor binding prediction and molecular graphs instead of
-questions that they didn't know to ask, it will be clear that deep learning has
-transformed biology and medicine.
+currently less developed than their supervised counterparts, but they may have
+the most potential because of how expensive and time-consuming it is to label
```
I know we've had / are having this discussion elsewhere, but unsupervised methods are valuable for many reasons, primary of which is that they are revealing the truth / otherwise unseen divisions rather than reinforcing a human-dictated classification. 

How to say this compactly is an open question. Perhaps in the later stages we need to look for all mentions of "unsupervised" and put in a single clear statement.

agitter commented:
Looks good.  I added a commit to fix one of the references.  If this builds okay, let's merge.
